internal/services/recoveryservices/azuresdkhacks/replicateditems_model.go

```diff
@@ -0,0 +1,8 @@
+package azuresdkhacks
+
+type DisableProtectionInput struct {
+	Properties EmptyInput `json:"properties"`
```

Comment:
Looking at the model here, since `ReplicationProviderInput` is a Discriminated Type that's Optional, as such this:

```
package main

import (
	"encoding/json"
	"log"

	"github.com/hashicorp/go-azure-sdk/resource-manager/recoveryservicessiterecovery/2022-10-01/replicationprotecteditems"
)

func main() {
	foo := replicationprotecteditems.DisableProtectionInput{}
	out, err := json.Marshal(foo)
	if err != nil {
		log.Fatalf("error: %+v", err)
	}
	log.Printf("JSON: %s", out)
}
```

Should output:

```
JSON: {"properties":{}}
```

But actually outputs:

```
JSON: {"properties":{"replicationProviderInput":null}}
```

As such this looks to be something we can workaround in `hashicorp/go-azure-sdk` - can you open an issue about this, with the full HTTP Requests/Responses so that we can diagnose this?

Reply:
here you go: https://github.com/hashicorp/go-azure-sdk/issues/387

Reply:
Just want to confirm if we need to wait for the `go-azure-sdk` workaround to merge this PR?